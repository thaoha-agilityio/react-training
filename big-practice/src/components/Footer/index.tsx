import { Link } from 'react-router-dom';

import { Box, Container, Divider, Flex, Heading, Input, Stack, Text } from '@chakra-ui/react';

// Constants
import { MENU, HELPS, PLACEHOLDER_MESSAGE } from '@constants';

const Footer = () => {
  return (
    <Box as='footer'>
      <Divider />
      <Container maxW='1240px' display='flex' gap='48px' flexDir='column' pt='48px' pb='35px'>
        <Flex
          justifyContent='space-between'
          fontSize={{ base: 'xs', md: 'sm' }}
          fontWeight='medium'
        >
          <Stack>
            <Heading fontSize={{ base: 'base', md: 'lg' }} fontWeight='bolder'>
              Funiro.
            </Heading>
            <Text color='gray.100' maxW='285px' mt='50px'>
              400 University Drive Suite 200 Coral Gables, FL 33134 USA
            </Text>
          </Stack>
          <Stack>
            <Text color='gray.100'>Links</Text>
            <Stack mt='55px' gap='46px'>
              {MENU.map((item) => (
                <Link to={item.path} key={item.title}>
                  {item.title}
                </Link>
              ))}
            </Stack>
          </Stack>
          <Stack>
            <Text color='gray.100'>Helps</Text>
            <Stack mt='55px' gap='46px'>
              {HELPS.map((item) => (
                <Link to={item.path} key={item.title}>
                  {item.title}
                </Link>
              ))}
            </Stack>
          </Stack>
          <Stack>
            <Text color='gray.100'>Newsletter</Text>
            <Flex mt='55px' gap='46px' flexDir={{ base: 'column', md: 'row' }}>
              <Box>
                <Input
                  placeholder={PLACEHOLDER_MESSAGE.NEWSLETTER}
                  border='none'
                  w={{ base: '100px', md: '200px' }}
                  h='24px'
                  pl='-1.5'
                  _focusVisible={{ border: 'none' }}
                />
                <Divider h='1px' color='gray.600' />
              </Box>
              <Box borderBottom='1px'>
                <Link to='/'>SUBSCRIBE</Link>
              </Box>
            </Flex>
          </Stack>
        </Flex>
        <Divider />
        <Text fontSize={{ base: 'xs', md: 'sm' }}>2023 furino. All rights reverved</Text>
      </Container>
    </Box>
  );
};

export default Footer;
