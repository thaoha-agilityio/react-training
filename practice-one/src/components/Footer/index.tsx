import './index.css';

const Footer = (): React.ReactElement => (
  <footer className="footer">
    <div className="about-web">
      <p className="web-info">
        &#169; 2022, Made with <span className="heart"> &hearts;</span> by
        <a href=""> nidves.com</a>
      </p>
      <p className="contact-web">
        <a href="">Documentation</a>
        <a href=""> Support</a>
      </p>
    </div>
  </footer>
);

export default Footer;
