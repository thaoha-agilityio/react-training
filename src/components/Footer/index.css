.footer {
  padding: 16px;
}
.about-web {
  display: flex;
  justify-content: space-between;
  line-height: calc(24 / 16);
}

.web-info {
  font-family: 'Inter';
  font-weight: 400;
  color: var(--color-primary-50);
}

.heart {
  color: var(--color-danger);
}

.web-info a {
  text-decoration: none;
  color: var(--color-secondary-100);
}

.contact-web a {
  text-decoration: none;
  font-family: 'Inter';
  color: var(--color-secondary-100);
}
