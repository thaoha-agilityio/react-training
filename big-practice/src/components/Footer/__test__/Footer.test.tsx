import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { PLACEHOLDER_MESSAGE } from '@constants';
import Footer from '../index';

describe('Footer Component', () => {
  it('renders the footer correctly', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Router>
        <Footer />
      </Router>,
    );

    const headingElement = getByText('Furniro.');
    expect(headingElement).toBeInTheDocument();

    const newsletterInput = getByPlaceholderText(PLACEHOLDER_MESSAGE.NEWSLETTER);
    const subscribeLink = getByText('SUBSCRIBE');
    expect(newsletterInput).toBeInTheDocument();
    expect(subscribeLink).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
