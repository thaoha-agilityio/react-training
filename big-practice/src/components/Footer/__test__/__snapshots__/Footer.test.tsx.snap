// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders the footer correctly 1`] = `
<div>
  <footer
    class="css-0"
  >
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1upb9tn"
    />
    <div
      class="chakra-container css-1mp8xin"
    >
      <div
        class="css-1rd3ca5"
      >
        <div
          class="chakra-stack css-8g8ihq"
        >
          <h2
            class="chakra-heading css-1anfozk"
          >
            Furniro.
          </h2>
          <p
            class="chakra-text css-nh0zjw"
          >
            400 University Drive Suite 200 Coral Gables, FL 33134 USA
          </p>
        </div>
        <div
          class="css-pdcctm"
        >
          <div
            class="chakra-stack css-1l051xc"
          >
            <p
              class="chakra-text css-14h5pyi"
            >
              Links
            </p>
            <div
              class="chakra-stack css-1enrxk9"
            >
              <a
                href="/"
              >
                Home
              </a>
              <a
                href="/shop"
              >
                Shop
              </a>
              <a
                href="/add-product"
              >
                Add Product
              </a>
            </div>
          </div>
          <div
            class="chakra-stack css-1l051xc"
          >
            <p
              class="chakra-text css-14h5pyi"
            >
              Helps
            </p>
            <div
              class="chakra-stack css-1enrxk9"
            >
              <a
                href="/"
              >
                Payment Options
              </a>
              <a
                href="/"
              >
                Returns
              </a>
              <a
                href="/"
              >
                Privacy Policies
              </a>
            </div>
          </div>
        </div>
        <div
          class="chakra-stack css-1umi0nh"
        >
          <p
            class="chakra-text css-14h5pyi"
          >
            Newsletter
          </p>
          <div
            class="css-1fb8qo7"
          >
            <div
              class="css-0"
            >
              <input
                class="chakra-input css-1alxois"
                placeholder="Enter Your Email Address"
              />
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-pio7g1"
              />
            </div>
            <div
              class="css-x39wmu"
            >
              <a
                href="/"
              >
                SUBSCRIBE
              </a>
            </div>
          </div>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <p
        class="chakra-text css-1il0tvo"
      >
        2023 furniro. All rights reverved
      </p>
    </div>
  </footer>
</div>
`;
